sys/src/9/port/sysfile.c

```diff
@@ -742,27 +742,24 @@ read(int fd, void *p, int32_t n, int64_t off)
 	 * The number of bytes read on this fd (c->offset) may be different
 	 * due to rewritings in mountfix.
 	 */
-		if(off == -1LL){	/* use and maintain channel's offset */
-			off = c->offset;
-		sequential = 1;
-	} else {
-		sequential = 0;
-	}
+	if(notpread)	/* use and maintain channel's offset */
+		off = c->offset;
+
 	if(c->qid.type & QTDIR){
 		/*
 		 * Directory read:
 		 * rewind to the beginning of the file if necessary;
 		 * try to fill the buffer via mountrockread;
-		 * set sequential to always maintain the Chan offset.
+		 * clear notpread to always maintain the Chan offset.
 		 */
 		if(off == 0LL){
-			if(sequential){
-				c->offset = 0;
-				c->devoffset = 0;
-			}
-			mountrewind(c);
-			unionrewind(c);
-		}
+ 			if(notpread){
```

Comment:
Looks like some random leading whitespace crept into this change?

Reply:
Cleaned!